dill/session.py

```diff
@@ -247,64 +346,60 @@ def dump_session(filename=str(TEMPDIR/'session.pkl'), main=None, byref=False, **
-            chunk = stream.read(n)
-            stream.seek(position)
-            return chunk
-        except (AttributeError, OSError):
-            raise NotImplementedError("stream is not peekable: %r", stream) from None
-
-def _make_peekable(stream):
-    """return stream as an object with a peek() method"""
-    import io
-    if hasattr(stream, 'peek'):
-        return stream
-    if not (hasattr(stream, 'tell') and hasattr(stream, 'seek')):
-        try:
-            return io.BufferedReader(stream)
-        except Exception:
-            pass
-    return _PeekableReader(stream)
-
 def _identify_module(file, main=None):
     """identify the name of the module stored in the given file-type object"""
-    from pickletools import genops
-    UNICODE = {'UNICODE', 'BINUNICODE', 'SHORT_BINUNICODE'}
-    found_import = False
+    import pickletools
+    NEUTRAL = {'PROTO', 'FRAME', 'PUT', 'BINPUT', 'MEMOIZE', 'MARK', 'STACK_GLOBAL'}
+    opcodes = ((opcode.name, arg) for opcode, arg, pos in pickletools.genops(file.peek(256))
+               if opcode.name not in NEUTRAL)
     try:
-        for opcode, arg, pos in genops(file.peek(256)):
-            if not found_import:
-                if opcode.name in ('GLOBAL', 'SHORT_BINUNICODE') and \
-                        arg.endswith('_import_module'):
-                    found_import = True
-            else:
-                if opcode.name in UNICODE:
-                    return arg
-        else:
-            raise UnpicklingError("reached STOP without finding main module")
+        opcode, arg = next(opcodes)
+        if (opcode, arg) == ('SHORT_BINUNICODE', 'dill._dill'):
+            # The file uses STACK_GLOBAL instead of GLOBAL.
+            opcode, arg = next(opcodes)
+        if not (opcode in ('SHORT_BINUNICODE', 'GLOBAL') and arg.split()[-1] == '_import_module'):
+            raise ValueError
+        opcode, arg = next(opcodes)
+        if not opcode in ('SHORT_BINUNICODE', 'BINUNICODE', 'UNICODE'):
+            raise ValueError
+        module_name = arg
+        if not (
+            next(opcodes)[0] in ('TUPLE1', 'TUPLE') and
+            next(opcodes)[0] == 'REDUCE' and
+            next(opcodes)[0] in ('EMPTY_DICT', 'DICT')
+        ):
+            raise ValueError
+        return module_name
+    except StopIteration:
+        raise UnpicklingError("reached STOP without finding module") from None
     except (NotImplementedError, ValueError) as error:
         # ValueError occours when the end of the chunk is reached (without a STOP).
         if isinstance(error, NotImplementedError) and main is not None:
             # file is not peekable, but we have main.
             return None
-        raise UnpicklingError("unable to identify main module") from error
+        raise UnpicklingError("unable to identify module") from error
+
+def is_pickled_module(filename, importable: bool = True) -> bool:
+    """Check if a file is a module state pickle file.
```

Comment:
It's not really transparent what `module state pickle file` means.  Your explanation in `Returns` is better.

Reply:
What about this for the short description:
> "Check if a file is a pickle file readable by `load_module()`."

Reply:
I like "Check if the pickle file was generates with `dump_module`".  You, could then, if you like, also add "and thus can be loaded with `load_module`".